src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/ParalogyFmsDTOValidator.java

```diff
@@ -20,57 +33,146 @@ public class ParalogyFmsDTOValidator {
-		// } else {
-		// ObjectResponse<NCBITaxonTerm> taxonResponse =
-		// ncbiTaxonTermService.getByCurie("NCBITaxon:" + dto.getSpecies());
-		// NCBITaxonTerm subjectTaxon = taxonResponse.getEntity();
-		// if (subjectTaxon == null) {
-		// paralogyResponse.addErrorMessage("Species",
-		// ValidationConstants.INVALID_MESSAGE + " (" + dto.getSpecies() + ")");
-		// } else if (!sameGenus(subjectTaxon, subjectGene.getTaxon())) {
-		// paralogyResponse.addErrorMessage("Species",
-		// ValidationConstants.INVALID_MESSAGE + " (" + dto.getSpecies() + ") for gene "
-		// + subjectGene.getCurie());
-		// }
-		// }
-		// }
-		// }
-		return null;
+		ObjectResponse<GeneToGeneParalogy> paralogyResponse = new ObjectResponse<GeneToGeneParalogy>();
+
+		String subjectGeneIdentifier = null;
+		String objectGeneIdentifier = null;
+
+		GeneToGeneParalogy paralogyData = null;
+
+		NCBITaxonTerm speciesTaxon = null;
+
+		if (StringUtils.isBlank(dto.getGene1())) {
+			paralogyResponse.addErrorMessage("gene1", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			subjectGeneIdentifier = convertToModCurie(dto.getGene1(), dto.getSpecies());
+		}
+		if (StringUtils.isBlank(dto.getGene2())) {
+			paralogyResponse.addErrorMessage("gene2", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			objectGeneIdentifier = convertToModCurie(dto.getGene2(), dto.getSpecies());
+		}
+
+		if (dto.getSpecies() == null) {
+			paralogyResponse.addErrorMessage("Species", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			ObjectResponse<NCBITaxonTerm> taxonResponse = ncbiTaxonTermService.getByCurie("NCBITaxon:" + dto.getSpecies());
+			speciesTaxon = taxonResponse.getEntity();
+			if (speciesTaxon == null) {
+				paralogyResponse.addErrorMessage("Species", ValidationConstants.INVALID_MESSAGE + " (" + dto.getSpecies() + ")");
+			}
+		}
+
+		Gene subjectGene = null;
+		if (StringUtils.isNotBlank(dto.getGene1())) {
+			subjectGene = geneService.findByIdentifierString(subjectGeneIdentifier);
+			if (subjectGene == null) {
+				paralogyResponse.addErrorMessage("gene1", ValidationConstants.INVALID_MESSAGE + " (" + subjectGeneIdentifier + ")");
+			} else {
+				if (!sameGenus(speciesTaxon, subjectGene.getTaxon())) {
+					paralogyResponse.addErrorMessage("Species", ValidationConstants.INVALID_MESSAGE + " (" + dto.getSpecies() + ") for gene " + subjectGene.getCurie());
+				}
+			}
+		}
+
+		Gene objectGene = null;
+		if (StringUtils.isNotBlank(dto.getGene2())) {
+			objectGene = geneService.findByIdentifierString(objectGeneIdentifier);
+			if (objectGene == null) {
+				paralogyResponse.addErrorMessage("gene2", ValidationConstants.INVALID_MESSAGE + " (" + objectGeneIdentifier + ")");
+			} else {
+				if (!sameGenus(speciesTaxon, objectGene.getTaxon())) {
+					paralogyResponse.addErrorMessage("Species", ValidationConstants.INVALID_MESSAGE + " (" + dto.getSpecies() + ") for gene " + objectGene.getCurie());
+				}
+			}
+		}
+
+		if (subjectGene != null && objectGene != null)
+
+		{
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.blocks.LeftCurlyCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'{' at column 3 should be on the previous line.